java/core/src/main/java/org/bondlib/StructBondType.java

```diff
@@ -175,14 +175,16 @@ protected abstract void deserializeStructFields(
         TaggedDeserializationContext context, TStruct value) throws IOException;
 
     /**
-     * Used by generated subclasses to deserialize declared fields of this struct, excluding inherited fields.
+     * Used by generated subclasses to deserialize fields of this struct using the runtime schema, excluding inherited
+     * fields.
      *
      * @param context contains the runtime context of the deserialization
+     * @param typeDef the TypeDef of this struct
      * @param value   the value to deserialize into
      * @throws IOException if an I/O error occurred
      */
     protected abstract void deserializeStructFields(
-        UntaggedDeserializationContext context, TStruct value) throws IOException;
+        UntaggedDeserializationContext context, TypeDef typeDef, TStruct value) throws IOException;
```

Comment:
>TypeDef typeDef,  [](start = 48, length = 17)

I believe this is cleaner with StructDef parameter. TypeDef is too generic and doesn't always map to a struct. Further, generated implementation only used the TypeDef to get StructDef from it.

Reply:
The current case uses the `StructDef`, but passing the `TypeDef` permits all cases (not just structs) to do error checking, right? That was the intent.

Reply:
This method is deserializeStructFields, which expects a struct as only structs can have fields. There is no scenario where TypeDef would be useful here.

Reply:
Done.